Sources/MapboxMaps/Style/Style.swift

```diff
@@ -989,6 +989,42 @@ public final class Style: StyleProtocol {
         return styleManager.getStyleAtmosphereProperty(forProperty: property)
     }
 
+    // MARK: - Models
+
+    /// Adds a `model` to the map
+    ///
+    /// - Parameters:
+    ///   - model: The model to add to the map.
+    ///   - Parameter identifier: A unique source identifier.
+    ///
+    /// - Throws: StyleError or type conversion errors
+    @_spi(Experimental) public func addModel(withId id: String, modelURI: String) throws {
+        try handleExpected {
+            styleManager.addStyleModel(forModelId: id, modelUri: modelURI)
+        }
+    }
+
+    /// Removes an existing style model.
+    ///
+    /// - Parameter id: Identifier of the style model to remove.
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    @_spi(Experimental) public func removeModel(withId id: String) throws {
+        try handleExpected {
+            styleManager.removeStyleModel(forModelId: id)
+        }
+    }
+
+    /// Checks whether a given style model exists.
+    ///
+    /// - Parameter id: Style model identifier.
+    ///
+    /// - Returns: `true` if the given model exists, `false` otherwise.
+    @_spi(Experimental) public func modelExists(withId id: String) -> Bool {
+        styleManager.hasStyleModel(forModelId: id)
+    }
+
```

Comment:
Do these replace model source?

Reply:
Good point.

> Do these replace model source?
Gl-native won't deprecate model source API and it will stay as is (_private_ API between mobile and native SDKs) since the API is designed around performant property update. At least for now.

> (and should Puck3D be updated to use these?)
These APIs allow instancing, by assigning 3D models to vector source's point geometry features. While feature state API  or updating geojson source allows updating individual instances, it is still expected to cost more that calling model source methods directly. It is experimental API and underlying implementation is expected to improve so we should revisit this and potential model source code removal.